tests/ga/test_extension.py

```diff
@@ -1379,149 +1381,6 @@ def test_ext_handler_io_error(self, mock_fileutil, *args):
-                with patch.object(protocol, "get_artifacts_profile",
-                                  return_value=InVMArtifactsProfile(json.dumps({'onHold': False}))):
-                    # 1 expected call count for Enable command
-                    assert_extensions_called(exthandlers_handler, expected_call_count=1)
-
-    def test_it_should_process_extensions_appropriately_on_artifact_hold(self, mock_get, mock_crypt, *args):
-        test_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE)
-        exthandlers_handler, protocol = self._create_mock(test_data, mock_get, mock_crypt, *args)
-
-        # enable extension handling blocking
-        conf.get_enable_overprovisioning = Mock(return_value=True)
-        mock_in_vm_artifacts_profile = InVMArtifactsProfile(MagicMock())
-
-        # Test when is_on_hold returns True - should not work
-        mock_in_vm_artifacts_profile.is_on_hold = Mock(return_value=True)
-        protocol.get_artifacts_profile = Mock(return_value=mock_in_vm_artifacts_profile)
-
-        exthandlers_handler.run()
-        exthandlers_handler.report_ext_handlers_status()
-
-        vm_agent_status = protocol.report_vm_status.call_args[0][0].vmAgent
-        self.assertEqual(vm_agent_status.status, "Ready", "Agent should report ready")
-        self.assertEqual(0, len(vm_agent_status.extensionHandlers),
-                         "No extensions should be reported as on_hold is True")
-        self.assertIsNone(vm_agent_status.vm_artifacts_aggregate_status.goal_state_aggregate_status,
-                          "No GS Aggregate status should be reported")
-
-        # Test when is_on_hold returns False - should work
-        mock_in_vm_artifacts_profile.is_on_hold = Mock(return_value=False)
-        protocol.get_artifacts_profile = Mock(return_value=mock_in_vm_artifacts_profile)
-        exthandlers_handler.run()
-        exthandlers_handler.report_ext_handlers_status()
-
-        self._assert_handler_status(protocol.report_vm_status, "Ready", 1, "1.0.0")
-        self.assertEqual("1", protocol.report_vm_status.call_args[0][
-            0].vmAgent.vm_artifacts_aggregate_status.goal_state_aggregate_status.in_svd_seq_no, "Incarnation mismatch")
-
-        # Test when in_vm_artifacts_profile is not available - Should work
-        protocol.get_artifacts_profile = Mock(return_value=None)
-        # Update GoalState
-        test_data.set_incarnation(2)
-        protocol.update_goal_state()
-
-        exthandlers_handler.run()
-        exthandlers_handler.report_ext_handlers_status()
-
-        self._assert_handler_status(protocol.report_vm_status, "Ready", 1, "1.0.0")
-        self.assertEqual("2", protocol.report_vm_status.call_args[0][
-            0].vmAgent.vm_artifacts_aggregate_status.goal_state_aggregate_status.in_svd_seq_no, "Incarnation mismatch")
-
-    def test_it_should_parse_valid_in_vm_metadata_properly(self, mock_get, mock_crypt, *args):
```

Comment:
These tests were replaced by new tests in ExtensionsGoalStateFromExtensionsConfigTestCase